### DISCUSSION TITLE
Project pull environment variable from another project?

### DISCUSSION THREAD

**Discussion by elvisun** [vercel/community#4644] [Help] [answered]
### Page to Investigate

https://vercel.com/_/__/settings/environment-variables

### Steps to Reproduce

Create 2 projects using the same git repo.

Then set different environment variables, see if one reuses the other's envvar.

I removed the env var from one project that's not using the correct envvar, my build fails saying there's no env var. Then when I add it back in it begins to use env var from another project

**Comment by amyegan** [comment]
I was not able to replicate this behavior. Are you able to share an [example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Reply by elvisun** [reply]
I tried again in a couple hours and it worked... so I can't 100% repo it, but it's really strange...

**Reply by sham601** [reply]
24 for hoes what is tha strange y

**Reply by sham601** [reply]
yea

**Comment by elvisun** [comment]
Hey Amy,

Ok I'm able to repo it again, I'm pretty confident there's something wrong at Vercel's level.

My setup is one git repo connected to 2 vercel projects.

1. ✅ If I deploy project B alone, it works.
2. ❌ If I deploy project A and then project B, project B would use project A's env var.
3. ✅ If I redeploy project B by clicking on the re-deploy button, project B will use project B's env var.

This basically breaks my production app on each git commit.

If you want send me an email I'll send you a loom recording with more details. 
`vercel-support-1` at `elvisun.com`

**Comment by elvisun** [comment]
Any updates? Still seeing this behavior consistently

**Comment by amyegan** [comment]
I'm still not able to replicate the problem with my basic test app. 

Do you happen to be using Nx or Turborepo? It's possible that this is a remote cache problem. The Turborepo docs have a [Configuring Cache Outputs](https://turbo.build/repo/docs/core-concepts/caching/what-to-cache#configuring-cache-outputs) page and the Nx docs have a [How Caching Works](https://nx.dev/concepts/how-caching-works#runtime-hash-inputs) page. Those pages should help if caching the source of the problem. 

If that's not it, I could use a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to dig into the problem further.

**Reply by elvisun** [reply, marked as answer]
Thank you Amy, I am using Nx so that is actually why. Can confirm ` --skip-nx-cache` fixed it
